### PR TITLE
Assignment 5 files

### PR DESCRIPTION
**Pull request by Swt16**

### CODE REVIEW

File: students/ShinTran/lesson05/mailroom3.py [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+        ["William Rodriguez", 41113.42],
+        ["Patricia Brown", 19184.81],
+        ["William Rodriguez", 51227.53],
+        ["Jennifer Miller", 53514.94],
+        ["Michael David", 31051.46],
+        ["Mary Johnson", 71646.16],
+        ["William Rodriguez", 69244.21],
+        ["Mary Johnson", 12689.07],
+        ["Michael David", 57145.50],
+        ["Patricia Brown", 81679.15],
```
Comment (charlesrobison) [inline students/ShinTran/lesson05/mailroom3.py]: Starting from lesson 4, your main data structure for donors should be a dictionary.  e.g., 

`donors = {
        "Jennifer Miller": [91561.25],
        "Mary Johnson": [5811.05]
}`
etc.

File: students/ShinTran/lesson05/mailroom3.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        '3': send_letters,
+        '4': exit_quit
+    }
+    try:
+        switch_dict.get(user_input)()
+    except (TypeError, ValueError):
+        print("Invalid input, {} please try again.".format(user_input))
+        action(main_prompt())
+    else:
+        action(main_prompt())
```
Comment (charlesrobison) [inline students/ShinTran/lesson05/mailroom3.py]: Good use of exception handling.

File: students/ShinTran/lesson05/mailroom3.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+    output.close()
+
+# Closes out of the prompt and ends the program
+def exit_quit():
+    quit()
+
+# Python program to use main for function call
+if __name__ == "__main__":
+    initialize()
+    action(main_prompt())
```
Comment (charlesrobison) [inline students/ShinTran/lesson05/mailroom3.py]: Not seeing use of comprehensions per the assignment requirements.

File: students/ShinTran/lesson05/except_exercise.py [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+
+# Finally, while still in the try/except block and regardless of whether
+# there were any exceptions, call the function last_fun with no
+# parameters. (pun intended)
+
+print()
+try:
+    last_fun()
+except:
+    print('Some error here.')
```
Comment (charlesrobison) [inline students/ShinTran/lesson05/except_exercise.py]: Good examples on this assignment!

File: students/ShinTran/lesson05/mailroom3.py [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+def send_letters():
+    global master_list
+    global name_list
+    donation_sum = 0
+    message = "Dear {:s},\n\
+    Thank you for the your generous total donation of ${:,.2f}.\n\
+    Sincerely,\n\
+    Your Local Charity"
+    for name in name_list:
+        output = open(name + ".txt",'w')
```
Comment (natasha-aleksandrova) [inline students/ShinTran/lesson05/mailroom3.py]: it is preferred to use `with` statement when working with files which takes care of making sure the file is closed.

**Pull request merged**